ydb/core/statistics/aggregator/tx_aggr_stat_response.cpp

```diff
@@ -0,0 +1,122 @@
+#include "aggregator_impl.h"
+
+#include <ydb/core/statistics/stat_service.h>
+
+namespace NKikimr::NStat {
+
+struct TStatisticsAggregator::TTxAggregateStatisticsResponse : public TTxBase {
+    NKikimrStat::TEvAggregateStatisticsResponse Record;
+
+    bool SendReqDistribution = false;
+    bool SendAggregate = false;
+
+    std::unique_ptr<TEvStatistics::TEvAggregateStatistics> Request;
+
+    TTxAggregateStatisticsResponse(TSelf* self, NKikimrStat::TEvAggregateStatisticsResponse&& record)
+        : TTxBase(self)
+        , Record(std::move(record))
+    {}
+
+    TTxType GetTxType() const override { return TXTYPE_AGGR_STAT_RESPONSE; }
+
+    bool Execute(TTransactionContext& txc, const TActorContext&) override {
+        SA_LOG_D("[" << Self->TabletID() << "] TTxAggregateStatisticsResponse::Execute");
+
+        ++Self->KeepAliveSeqNo; // cancel timeout events
+
+        NIceDb::TNiceDb db(txc.DB);
+
+        for (auto& column : Record.GetColumns()) {
+            auto tag = column.GetTag();
+            for (auto& statistic : column.GetStatistics()) {
+                if (statistic.GetType() == NKikimr::NStat::COUNT_MIN_SKETCH) {
+                    if (Self->ColumnNames.find(tag) == Self->ColumnNames.end()) {
+                        continue;
+                    }
+                    if (Self->CountMinSketches.find(tag) == Self->CountMinSketches.end()) {
+                        Self->CountMinSketches[tag].reset(TCountMinSketch::Create());
+                    }
+
+                    auto* data = statistic.GetData().Data();
+                    auto* sketch = reinterpret_cast<const TCountMinSketch*>(data);
+                    auto& current = Self->CountMinSketches[tag];
+                    *current += *sketch;
+                }
+            }
+        }
+
+        if (Record.FailedTabletsSize() == 0 ||
+            Self->TraversalRound >= Self->MaxTraversalRoundCount)
+        {
+            Self->SaveStatisticsToTable();
+            return true;
+        }
+
+        std::unordered_map<ui32, std::unordered_set<ui64>> nonLocalTablets;
```

Comment:
It feels like failed tablets should be unique. Is it?
In such case vector instead of unordered_set looks better